### PR TITLE
[Inductor] Kill has_aliasing

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112925
* __->__ #112875

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -4595,11 +4588,12 @@ def get_unbacked_symbol_uses(self):
     def should_allocate(self):
         return False
 
-    def has_aliasing(self):
-        return any(
-            isinstance(inp, (FallbackKernel, ComplexView)) and inp.has_aliasing()
+    def get_alias_names(self):
+        return [
+            inp.get_name()
```
Comment (Chillee) [inline torch/_inductor/ir.py:4593]: I think this is probably more clear as `inp.get_alias_names()` without the `isinstance` check?

Comment (Chillee) [reply torch/_inductor/ir.py:4593]: But actually, I also do wonder whether this should even have `get_alias_names` set on it? It doesn't actually represent any buffer, so why should it alias anything?

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -4315,11 +4309,13 @@ def has_side_effects(self):
...
 
-    def has_aliasing(self):
+    def get_alias_names(self):
         # TODO - some fallbacks are still OpOverloadPackets
         if not isinstance(self.op_overload, torch._ops.OpOverload):
-            return False
-        return torch._inductor.utils.is_view(self.op_overload)
+            return []
+        if torch._inductor.utils.is_view(self.op_overload):
+            return [inp.get_name() for inp in self.inputs]
```
Comment (Chillee) [inline torch/_inductor/ir.py:4317]: This is a bit weird to me - we're not actually aliasing all the inputs, presumably only one of them.

Moreover, this breaks our invariant that `get_alias_names()` always has length <= 1.

File: torch/_inductor/ir.py [hunk 3]
```diff
@@ -4315,11 +4309,13 @@ def has_side_effects(self):
...
         return get_schema_info(self.op_overload).is_mutable()
 
-    def has_aliasing(self):
+    def get_alias_names(self):
         # TODO - some fallbacks are still OpOverloadPackets
         if not isinstance(self.op_overload, torch._ops.OpOverload):
-            return False
-        return torch._inductor.utils.is_view(self.op_overload)
+            return []
+        if torch._inductor.utils.is_view(self.op_overload):
```
Comment (eellison) [inline torch/_inductor/ir.py:4316]: We could be more exact here but it's probably not needed.

**Review by Chillee — APPROVED** [review]
Mostly LGTM

File: torch/_inductor/ir.py (hunk 2 above)
Comment (oulgen) [reply torch/_inductor/ir.py:4317]: We don't have that invariant -- that's only for get_mutation_names

File: torch/_inductor/ir.py [hunk 4]
```diff
@@ -4595,11 +4588,12 @@ def get_unbacked_symbol_uses(self):
...
 
-    def has_aliasing(self):
-        return any(
-            isinstance(inp, (FallbackKernel, ComplexView)) and inp.has_aliasing()
+    def get_alias_names(self):
+        return [
+            inp.get_name()
             for inp in self.inputs
-        )
+            if isinstance(inp, (FallbackKernel, ComplexView))
```
Comment (eellison) [inline torch/_inductor/ir.py:4595]: Don't we want 
```if isinstance(inp, (FallbackKernel, ComplexView)) and len(inp.get_alias_names()) != 0```

**Comment by oulgen** [comment]
@pytorch merge

**Pull request closed without merging**